packages/svelte/src/compiler/phases/3-transform/utils.js

```diff
@@ -270,21 +270,28 @@ export function clean_nodes(
 
 	var first = trimmed[0];
 
-	/**
-	 * In a case like `{#if x}<Foo />{/if}`, we don't need to wrap the child in
-	 * comments — we can just use the parent block's anchor for the component.
-	 * TODO extend this optimisation to other cases
-	 */
-	const is_standalone =
-		trimmed.length === 1 &&
-		((first.type === 'RenderTag' && !first.metadata.dynamic) ||
-			(first.type === 'Component' &&
-				!state.options.hmr &&
-				!first.attributes.some(
-					(attribute) => attribute.type === 'Attribute' && attribute.name.startsWith('--')
-				)));
-
-	return { hoisted, trimmed, is_standalone };
+	return {
+		hoisted,
+		trimmed,
+		/**
+		 * In a case like `{#if x}<Foo />{/if}`, we don't need to wrap the child in
+		 * comments — we can just use the parent block's anchor for the component.
+		 * TODO extend this optimisation to other cases
+		 */
+		is_standalone:
```

Comment:
I'd love a test case that captured these different edge cases and prevents us breaking something in future in case we tweak some of these.

Reply:
Agree, though it turns out to be a slightly bigger change, because at the moment our hydration tests are forgiving about things like additional comments. Will look into whether it's practical to change it within this PR